src/renderer/components/+preferences/add-helm-repo-dialog.tsx

```diff
@@ -0,0 +1,171 @@
+import "./add-helm-repo-dialog.scss";
+
+import React from "react";
+import { remote } from "electron";
+import { observable } from "mobx";
+import { observer } from "mobx-react";
+import { t, Trans } from "@lingui/macro";
+import { _i18n } from "../../i18n";
+import { Dialog, DialogProps } from "../dialog";
+import { Wizard, WizardStep } from "../wizard";
+import { Input } from "../input";
+import { Checkbox } from "../checkbox";
+import { Button } from "../button";
+import { systemName, isUrl, isPath } from "../input/input_validators";
+import { SubTitle } from "../layout/sub-title";
+import { Icon } from "../icon";
+import { Notifications } from "../notifications";
+import { HelmRepo, repoManager } from "../../../main/helm/helm-repo-manager";
+
+interface Props extends Partial<DialogProps> {
+  onAddRepo: Function
+}
+
+enum FileType {
+  CaFile = "caFile",
+  KeyFile = "keyFile",
+  CertFile = "certFile",
+}
+
+@observer
+export class AddHelmRepoDialog extends React.Component<Props> {
+  private emptyRepo = {name: "", url: "", username: "", password: "", insecureSkipTlsVerify: false, caFile:"", keyFile: "", certFile: ""};
+
+  @observable static isOpen = false;
+
+  static open() {
+    AddHelmRepoDialog.isOpen = true;
+  }
+
+  static close() {
+    AddHelmRepoDialog.isOpen = false;
+  }
+
+  @observable helmRepo : HelmRepo = this.emptyRepo;
+  @observable showOptions = false;
+
+  close = () => {
+    AddHelmRepoDialog.close();
+    this.helmRepo = this.emptyRepo;
+    this.showOptions = false;
+  };
+
+  setFilepath(type: FileType, value: string) {
+    this.helmRepo[type] = value;
+  }
+
+  getFilePath(type: FileType) : string {
+    return this.helmRepo[type];
+  }
+
+  async selectFileDialog(type: FileType) {
+    const { dialog, BrowserWindow } = remote;
+    const { canceled, filePaths } = await dialog.showOpenDialog(BrowserWindow.getFocusedWindow(), {
```

Comment:
Does our dialog need to filter extensions for key files? If so, please provide a list of them.
See https://github.com/lensapp/lens/blob/master/src/renderer/components/+extensions/extensions.tsx#L149 for reference.

Reply:
Hmm, checked helm code and didn't find any file extension checks. Added some default lists of extensions:
![image](https://user-images.githubusercontent.com/53330707/101156860-a772eb80-3642-11eb-927e-48f2ddd501c3.png)
Is it ok? 
Also, I've added "Any" filter in every file dialog

Reply:
Looks good and very detailed 😄

Reply:
Just the laziest way to set extensions :) Otherwise, I'll need  to convert FileType to a separate class, and I'm not sure in profit of this

Reply:
Just took from [this](https://doc-archives.microstrategy.com/producthelp/10.10/SystemAdmin/WebHelp/Lang_1033/Content/certificate_file_overview.htm) :)